src/iso9660.rs

```diff
@@ -48,8 +48,15 @@ pub struct IsoFs {
 
 impl IsoFs {
     pub fn from_file(mut file: fs::File) -> Result<Self> {
+        let length = file.metadata()?.len();
         let descriptors = get_volume_descriptors(&mut file)?;
-        Ok(Self { descriptors, file })
+        let iso_fs = Self { descriptors, file };
+        let primary = iso_fs.get_primary_volume_descriptor()?;
+        if primary.volume_space_size * ISO9660_SECTOR_SIZE as u64 > length {
```

Comment:
I think in the case of a corrupted ISO this multiplication could overflow?  Probably worth something like:

```suggestion
    if primary.volume_space_size.checked_mul(ISO9660_SECTOR_SIZE as u64).ok_or_else(|| anyhow!("Invalid ISO volume size"))? > length 
```
or so?

Reply:
`volume_space_size` is 32 bits on disk, so this can't overflow.